client/ociindex/ociindex.go

```diff
@@ -155,6 +155,9 @@ func insertDesc(index *ocispecs.Index, desc ocispecs.Descriptor, tag string) err
 	if index.SchemaVersion == 0 {
 		index.SchemaVersion = 2
 	}
+	if index.MediaType == "" {
```

Comment:
While this matches the `SchemaVersion` check, it is hard to understand why this is inside `insertDesc` function where it behaves like a side-effect. iiuc better to move both of these zero value initializations outside of the `insertDesc` function.